src/mongo/tree/MongoDocumentTreeItem.ts

```diff
@@ -58,16 +58,14 @@ export class MongoDocumentTreeItem implements IAzureTreeItem {
     }
 
     public async update(newDocument: IMongoDocument): Promise<IMongoDocument> {
-        if (Object.getOwnPropertyNames(newDocument).length === 0) {
-            throw new Error("Updating a document with no fields in the document does nothing");
-        }
-        else {
-            this.document = await MongoDocumentTreeItem.update(this._collection, newDocument);
-            return this.document;
-        }
+        this.document = await MongoDocumentTreeItem.update(this._collection, newDocument);
+        return this.document;
     }
 
     public static async update(collection: Collection, newDocument: IMongoDocument): Promise<IMongoDocument> {
+        if (Object.getOwnPropertyNames(newDocument).indexOf("_id") < 0) {
```

Comment:
suggest: 
  if ("_id" in newDocument)

or

const id = newDocument._id;
if (!id) {
   throw ...
}
const filter: object = { _id: id };

Reply:
Note that the last one will throw if "_id" exists but is empty.  The other checks you've got will pass if _id is an empty string (or if it's a truthy non-string).